src/command.js

```diff
@@ -23,55 +23,80 @@ export function matchImageSnapshotCommand(defaultOptions) {
 
     const startTime = Date.now();
 
+    let cypressContext = Cypress.mocha.getRunner().suite.ctx.test;
+    let cypressPath = [cypressContext.title];
+    let rootCypressContext = cypressContext;
+    while (rootCypressContext.parent != null) {
+      rootCypressContext = rootCypressContext.parent;
+      cypressPath.push(rootCypressContext.title);
+    }
+
+    var name = typeof maybeName === 'string' ? maybeName : undefined;
+    const rawSnapshotFilepath = cypressContext.titlePath();
+    const snapshotFilepath = rawSnapshotFilepath.map(str =>
+      str.replace(/[/\\?%*:| "<>]/g, '-').toLowerCase()
+    );
+    if (name == undefined && snapshotFilepath.length > 0) {
+      name = snapshotFilepath.pop();
+    }
+    let forceWaiting = false;
+
     const match = () => {
       cy.task(MATCH, {
         screenshotsFolder,
         updateSnapshots,
+        snapshotFilepath,
+        snapshotName: name,
         options,
-      });
+      }).then(ret => {
+        forceWaiting = !ret.snapshotExists;
+        if (forceWaiting) {
+          // if we are
```

Comment:
This is not a useful comment :)

Reply:
Hehe, something distracted me mid-sentence :D